backend/src/vault/vault-token.js

```diff
@@ -0,0 +1,50 @@
+import { getAuthenticator } from './vault-custom'
+import rp from 'request-promise'
+import url from 'url'
+import config from '../config'
+
+/**
+ * If the defined TTL of the vault token is expired, it will try to renew it
+ * @return {function} vault token renew middleware
+ */
+export default function vaultTokenRenew() {
+  return function(req, res, next) {
+    const timeElapsed = new Date().getTime() - req.session.vaultTokenTTL
+    const vaultService = config.services.find(s => s.type === 'vault')
+    if (timeElapsed >= vaultService.ttl) {
+      try {
+        renewToken(req.session.vaultToken, vaultService)
+          .then(() => {
+            req.session.vaultTokenTTL = new Date().getTime()
+            req.session.update()
+          })
+          .catch(function(err) {
+            next(err)
```

Comment:
If you're calling `next` in `catch`, you don't need `finally` anymore. (You'd call `next` twice if an error occurs).